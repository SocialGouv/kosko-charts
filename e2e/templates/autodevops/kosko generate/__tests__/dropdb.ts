//

import { config } from "dotenv";
import { template, TIMEOUT } from "e2e/templates/helpers";
import execa from "execa";
import { basename, resolve } from "path";

//

const cwd = template(basename(resolve(__dirname, "..", "..")));

test(
  "yarn --silent dropdb : should generate a dropdb manifest",
  async () => {
    const gitlabEnv = config({
      path: resolve(cwd, "./environments/.gitlab-ci.env"),
    }).parsed;
    // Required to allow seemless integration code example
    const result = await execa("yarn", ["--silent", "dropdb"], {
      cwd,
      env: { ...gitlabEnv },
    });

    expect(result.stdout).toMatchSnapshot();
    expect(result.exitCode).toEqual(0);
  },
  TIMEOUT
);
