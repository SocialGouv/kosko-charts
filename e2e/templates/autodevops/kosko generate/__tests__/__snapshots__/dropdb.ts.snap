// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yarn --silent dropdb : should generate a dropdb manifest 1`] = `
"apiVersion: batch/v1
kind: Job
metadata:
  annotations:
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/nonce: ''
    kapp.k14s.io/update-strategy: fallback-on-replace
  name: drop-azure-db-8843083e
spec:
  backoffLimit: 0
  template:
    metadata:
      annotations:
        kapp.k14s.io/deploy-logs: ''
    spec:
      containers:
        - command:
            - drop-db-user
          env:
            - name: DROP_DATABASE
              value: autodevops_e2ebranch42
            - name: DROP_USER
              value: user_e2ebranch42
          envFrom:
            - secretRef:
                name: azure-pg-admin-user
          image: >-
            registry.gitlab.factory.social.gouv.fr/socialgouv/docker/azure-db:2.6.1
          imagePullPolicy: IfNotPresent
          name: drop-db-user
          resources:
            limits:
              cpu: 300m
              memory: 256Mi
            requests:
              cpu: 100m
              memory: 64Mi
      restartPolicy: Never
  ttlSecondsAfterFinished: 86400"
`;
